src/housekeeping/hatpro.py

```diff
@@ -103,6 +103,8 @@ def _read_data(self, file: BinaryIO):
 
 class HatproHkdNc:
     def __init__(self, filename: str | bytes | PathLike):
+        if isinstance(filename, bytes):
+            filename = filename.decode()
```

Comment:
Yksinkertaisempi näin:

```suggestion
    def __init__(self, filename: str | PathLike):
```